mypyc/lib-rt/misc_ops.c

```diff
@@ -294,6 +294,12 @@ PyObject *CPyType_FromTemplate(PyTypeObject *template_,
     return NULL;
 }
 
+PyObject *CPyType_FromTemplateWarpper(PyObject *template_,
+                                      PyObject *orig_bases,
+                                      PyObject *modname) {
+    return CPyType_FromTemplate((PyTypeObject *)template_, orig_bases, modname);
+}
```

Comment:
There's no need for a wrapper function, since the function we are wrapping is also defined here. You can just move the cast to the implementation of `CPyType_FromTemplate`, and this wrapper functions becomes redundant.

Reply:
Unfortunately, that would break one other existing code from `generate_module_def` in `emitmodule.py` , it uses:
```python
emitter.emit_lines(
  '{t} = (PyTypeObject *)CPyType_FromTemplate({t}_template, NULL, modname);'.
  format(t=type_struct))
```
if we remove the wrapper function and change the signature, we'll need to change the emit here.

Reply:
Hmm maybe we should change that emit then (add a cast?). This way we'd only need a single function.

Reply:
OK I'll patch that also in this PR.